### ISSUE TITLE
Add back support for different input types

### ISSUE THREAD

**Issue by dmehra** [juttle/outrigger#27]
Outrigger currently supports inputs of type "text" and "number". The full list we want to support is:

- [x] text
- [x] number
- [ ] date
- [ ] duration
- [ ] filter

The input control widget that works now is an input box. The full list we wish to support is:

- [x] input box
- [ ] dropdown
- [ ] combobox

There is a separate issue for supporting juttle-powered inputs.

**Comment by go-oleg** [comment]
@demmer and I came up with the following proposal for which input types to implement.

Supported input types:
- [x] text
- [x] number
- [x] date
- [x] duration
- [ ] filter
- [x] select
- [x] multiselect

Just like in Jut 1.0, the type of input you choose to use will drive the UI component that renders it. For example, a `text` input will render as an [input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input) with `type="text"`. A `select` will render as a [select](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select)-like component.

Differences from Jut 1.0
- Get rid of distinction between `dropdown` and `combobox` and fold them into a `select`. This UI component will behave mostly like the previous `combobox` w.r.t. type-ahead, but it will not allow you to choose a value that is not in the list
- Break the `-multi true` option of the original `combobox` into its own input type called `multiselect`.

**Comment by dmajda** [comment]
> Differences from Jut 1.0
>  - Get rid of distinction between `dropdown` and `combobox` and fold them into a `select`. This UI component will behave mostly like the previous `combobox` w.r.t. type-ahead, but it will not allow you to choose a value that is not in the list
>  - Break the `-multi true` option of the original `combobox` into its own input type called `multiselect`.

Could you share reasons for these differences? (Not opposing, just curious if they are what I think they are :-)

**Comment by go-oleg** [comment]
The distinction between `dropdown` and `combobox` was always a bit ambiguous for users.

- We decided to go with `combobox` over `dropdown` because filtering behavior it provides is easier to discover than `dropdown`'s jump-to-match behavior.
- We got rid of the "choose non-existant value" because that just doesn't seem to match the idea of a `select` list where you are expected to select an item form an enumerated list.
- Lastly, we broke up `select` into `select` and `multiselect` to handle multiple selection because that makes it easier to be explicit about the expected return values of each input type. A `multiselect` will always return an Array whereas a `select` will most often return a scalar value (though it could return an Array if a user chose to inject Arrays as the `value` properties of the `items`).
